vllm/model_executor/layers/mamba/ops/mamba_ssm.py

```diff
@@ -301,11 +303,39 @@ def selective_scan_fn(u,
                       z=None,
                       delta_bias=None,
                       delta_softplus=False,
-                      return_last_state=False,
-                      position_indices=None,
-                      prev_state=None):
-    """if return_last_state is True, returns (out, last_state)
-    last_state has shape (batch, dim, dstate). 
+                      cu_seq_len=None,
+                      cache_indices=None,
+                      has_initial_state=None,
+                      ssm_states=None) -> Tuple[torch.Tensor, torch.Tensor]:
+    """
+    u: (dim, cu_seq_len) for varlen or (batch, dim, seqlen) 
+    delta: (dim, cu_seq_len) for varlen or (batch, dim, seqlen)
+    A: (dim, dstate) 
+    B: (ngroups, dstate, cu_seq_len) for varlen or (batch,ngroups,dstate,seqlen)
+    C: (ngroups, dstate, cu_seq_len) for varlen or (batch,ngroups,dstate,seqlen)
+    D: (dim,) 
+    z: (dim, cu_seq_len) for varlen or (batch, dim, seqlen) 
+    dt_bias: (dim,) or (dim)
+    cu_seq_len: (batch)
+        Cumulative tokens along the last dimension, 
+        sequence lengths are passed through cu_seq_len therefore are required 
+        for variable lengths kernel activation.
+        for example: cu_seq_len = torch.Tensor([10,15,16]) 
+        then u.shape = (dim,16)
+    cache_indices: (batch)
+        A tensor with each cell is a correspondent 
+        input and output ssm_state index
+    has_initial_state: (batch)
+        A tensor populated with ones and zeros, 
+        indicate if the ssm_state at the corresponding index should be 
+        used as initial state. Not providing argument assumes 
+        there's no initial state
```

Comment:
I noticed that `has_initial_state` is created with a scalar type of int32. Should we use `dtype=torch.bool` instead?

Reply:
Addressed